File: tests/06_phases_openmp.dg/nanox/cxx/success_task_01.cpp
/*
<testinfo>
test_generator=config/mercurium-nanox
</testinfo>
*/

template <typename T>
void copy(T* dst, T* src, int N)
{
    for(int i = 0; i < N; i++)
    {
#pragma omp task shared(dst, src)
        dst[i] = src[i];
    }
#pragma omp taskwait
}

void f(void)
{
    int a[10], b[10];

    copy(a, b, 10);
}

File: tests/06_phases_openmp.dg/common/fortran/success_task_01.f90
! <testinfo>
! test_generator=config/mercurium-nanox
! </testinfo>
PROGRAM P
    INTEGER :: X

    X = 1

    !$OMP TASK SHARED(X)
        X = 3
    !$OMP END TASK

    !$OMP TASKWAIT

    IF (X /= 3) STOP "ERROR"
END PROGRAM P
